hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -2483,6 +2483,36 @@ public char[] getPasswordFromCredentialProviders(String name)
     return pass;
   }
 
+  /**
+   * Try and resolve the provided element name as a credential provider
+   * alias from the given provider path.
+   * @param name alias of the provisioned credential
+   * @param name credProviderPath path for credential provider
+   * @return password or null if not found
+   * @throws IOException when error in fetching password
+   */
+  public char[] getPasswordFromCredentialProvider(String name, String credProviderPath)
```

Comment:
have caller pass in the URI; makes it their problem to create that from a Path, string, etc. change name appropriately